src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
+
+  return sum / women.length;
 }
```

Comment:
Same as in **calculateMenAverageAge** ternary operator here is in my opinion harder to read than if-else, sum could be also calculated using reduce and division by 0 is not handled correctly.
```js
function calculateWomenAverageAge(people, withChildren) {
  let women = [];

  if (withChildren) {
    women = people.filter((p) => p.sex === 'f' && people.find((p1) => p1.mother === p.name));
  } else {
    women = people.filter((p) => p.sex === 'f');
  }

  if (women.length === 0) {
    return 0; // Handle division by zero
  }

  const sum = women.reduce((acc, w) => acc + (w.died - w.born), 0);

  return sum / women.length;
}
```